keps/sig-storage/2857-runtime-assisted-pv-mounts/README.md

```diff
@@ -0,0 +1,1767 @@
+expanding the file system when requested) with the container runtime handler
+based on the file system deferral strategy specified using the new parameters
+above. If a CSI node plugin cannot use the deferral strategy that is passed, it
+is expected to return an `Unimplemented` error.
+
+### A mutating webhook to set FSDeferralStrategy on Pods
+
+A general purpose mutating webhook named `fs-deferral-strategy-assigner.k8s.io`
+can be installed in a cluster to automatically set `FSDeferralStrategy` in the
+`PersistentVolumeClaimVolumeSource`, `EphemeralVolumeSource` or
+`CSIVolumeSource` structs in a pod spec when certain conditions are met. The
+section below specifies one possible approach for a contaienr runtime and CSI
+plugin to surface capabilities to the mutating webhook using annotations. It's
+possible to have a different channel to feed this information to the webhook,
+for example through a configmap. The webhook will evaluate the following:
+
+- The capability of a container runtime (associated with the RuntimeClass
+specified by a pod) to execute file system mount, post mount configuration and
+management operations for a specific set of file systems when directed using a
+supported file system deferral strategy. The capabilities can be expressed as
+the following annotations on the RuntimeClass associated with the container
+runtime:
+`runtime.k8s.io/fs-deferral-strategy`,
+`mount.capabilities.runtime.k8s.io/supports-fsgroup`,
+`mount.capabilities.runtime.k8s.io/supports-fsgroup-change-policy`,
+`mount.capabilities.runtime.k8s.io/supports-subpath` and
+`mount.capabilities.runtime.k8s.io/supports-filesystems`.
+
+- The capability of a CSI plugin to defer file system mount, post mount
+configuration and management operations using a strategy supported by a runtime.
+The capabilities can be expressed as a set of annotations on the CSIDriver
+associated with the CSI plugin associated with either inline CSI volumes or
+storage classes specified by PVCs or generic ephemeral volumes in a
+pod's volumes section. The capabilities may also be expressed as a set of
+annotations on the StorageClass associated with a PVC/EphemeralVolume. The
+latter applies to scenarios where a storage class is associated with a
+provisioner that may be different from a CSI node plugin handling volumes
+associated with the storage class. If a storage class refers to a CSI plugin
+name as the provisioner and specifies capabilities also specified by the
+CSIDriver, the capabilities specified by the storage class annotations take
+priority. This allows the capabilities of the CSI driver around file system
+deferral to be overridden/masked for volumes associated with a certain storage
+class. The following annotations on CSIDriver and StorageClass can be used to
+express (or mask) capabilities around deferral of file system operations:
+`csi.storage.k8s.io/fs-deferral-strategy`,
+`csi.storage.k8s.io/supports-filesystems`,
```

Comment:
Is the list of filesystems limited to filesystems on a block device (say ext4) or can it have also shared filesystems (like nfs)?

Reply:
Scope overall is limited to filesystems on a block device like ext4 and xfs.